tests/std/tests/P0881R7_stacktrace/test.cpp

```diff
@@ -0,0 +1,278 @@
+    assert(filesystem::path(all_but_top[2].source_file()).filename() == ""sv);
+
+    assert(all_but_top[0].source_line() == 0);
+    assert(all_but_top[1].source_line() == 0);
+    assert(all_but_top[2].source_line() == 0);
+#endif // ^^^ !HAS_DEBUG_INFO ^^^
+
+#ifdef HAS_NAMES
+    assert(trim_past_plus(all_but_top[0].description()) == "P0881R7_stacktrace!all_but_top_inner"sv);
+    assert(trim_past_plus(all_but_top[1].description()) == "P0881R7_stacktrace!all_but_top_outer"sv);
+    assert(trim_past_plus(all_but_top[2].description()) == "P0881R7_stacktrace!all_but_top_outermost"sv);
+#else // ^^^ HAS_NAMES ^^^ / vvv !HAS_NAMES vvv
+    assert(trim_past_plus(all_but_top[0].description()) == "P0881R7_stacktrace"sv);
+    assert(trim_past_plus(all_but_top[1].description()) == "P0881R7_stacktrace"sv);
+    assert(trim_past_plus(all_but_top[2].description()) == "P0881R7_stacktrace"sv);
+#endif // ^^^ !HAS_NAMES ^^^
+
+    auto three_excluding_top = three_excluding_top_outermost();
+    assert(three_excluding_top.size() == 3);
+#ifdef HAS_DEBUG_INFO
+    assert(filesystem::path(three_excluding_top[0].source_file()).filename() == "test.cpp"sv);
+    assert(filesystem::path(three_excluding_top[1].source_file()).filename() == "test.cpp"sv);
+    assert(filesystem::path(three_excluding_top[2].source_file()).filename() == "test.cpp"sv);
+
+    assert(three_excluding_top[0].source_line() == base_line + 41);
+    assert(three_excluding_top[1].source_line() == base_line + 45);
+    assert(three_excluding_top[2].source_line() == base_line + 49);
+#else // ^^^ HAS_DEBUG_INFO ^^^ / vvv !HAS_DEBUG_INFO vvv
+    assert(filesystem::path(three_excluding_top[0].source_file()).filename() == ""sv);
+    assert(filesystem::path(three_excluding_top[1].source_file()).filename() == ""sv);
+    assert(filesystem::path(three_excluding_top[2].source_file()).filename() == ""sv);
+
+    assert(three_excluding_top[0].source_line() == 0);
+    assert(three_excluding_top[1].source_line() == 0);
+    assert(three_excluding_top[2].source_line() == 0);
+#endif // ^^^ !HAS_DEBUG_INFO ^^^
+
+#ifdef HAS_NAMES
+    assert(trim_past_plus(three_excluding_top[0].description()) == "P0881R7_stacktrace!three_excluding_top_inner"sv);
+    assert(trim_past_plus(three_excluding_top[1].description()) == "P0881R7_stacktrace!three_excluding_top_outer"sv);
+    assert(
+        trim_past_plus(three_excluding_top[2].description()) == "P0881R7_stacktrace!three_excluding_top_outermost"sv);
+#else // ^^^ HAS_NAMES ^^^ / vvv !HAS_NAMES vvv
+    assert(trim_past_plus(three_excluding_top[0].description()) == "P0881R7_stacktrace"sv);
+    assert(trim_past_plus(three_excluding_top[1].description()) == "P0881R7_stacktrace"sv);
+    assert(trim_past_plus(three_excluding_top[2].description()) == "P0881R7_stacktrace"sv);
+#endif // ^^^ !HAS_NAMES ^^^
+
+    try {
+        (void) all.at(all.size());
+        assert(false); // should have thrown
+    } catch (out_of_range) {

```

Comment:
Should include `<stdexcept>`. Also, `const out_of_range&` would avoid any concerns about copying exceptions.